server/service/src/processors/transfer/shipment/test.rs

```diff
@@ -786,31 +964,289 @@ impl ShipmentTransferTester {
             .unwrap();
 
         assert!(outbound_shipment.is_some());
-        check_shipment_status(
+        check_invoice_status(
             &outbound_shipment.unwrap(),
             &self.inbound_shipment.clone().unwrap(),
         )
     }
+
+    pub(crate) fn insert_outbound_return(&self, connection: &StorageConnection) {
+        let inbound_shipment_id = self.inbound_shipment.clone().map(|r| r.id);
+        insert_extra_mock_data(
+            connection,
+            inline_init(|r: &mut MockData| {
+                r.invoices = vec![inline_edit(&self.outbound_return, |mut r| {
+                    r.original_shipment_id = inbound_shipment_id;
+                    r
+                })];
+                r.invoice_lines = vec![self.outbound_return_line.clone()]
+            })
+            .join(self.extra_mock_data.clone()),
+        );
+    }
+
+    pub(crate) fn check_inbound_return_not_created(&self, connection: &StorageConnection) {
+        assert_eq!(
+            InvoiceRepository::new(connection).query_one(
+                InvoiceFilter::new_match_linked_invoice_id(&self.outbound_return.id)
+            ),
+            Ok(None)
+        )
+    }
+
+    pub(crate) fn update_outbound_return_to_picked(&mut self, service_provider: &ServiceProvider) {
+        let ctx = service_provider
+            .context(self.inbound_store.id.clone(), "".to_string())
+            .unwrap();
+        self.outbound_return = service_provider
+            .invoice_service
+            .update_outbound_return(
+                &ctx,
+                inline_init(|r: &mut UpdateOutboundReturn| {
+                    r.outbound_return_id = self.outbound_return.id.clone();
+                    r.status = Some(UpdateOutboundReturnStatus::Picked);
+                }),
+            )
+            .unwrap()
+            .invoice_row;
+    }
+
+    pub(crate) fn check_inbound_return_created(&mut self, connection: &StorageConnection) {
+        let inbound_return = InvoiceRepository::new(connection)
+            .query_one(InvoiceFilter::new_match_linked_invoice_id(
+                &self.outbound_return.id,
+            ))
+            .unwrap();
+
+        assert!(inbound_return.is_some());
+        let inbound_return = inbound_return.unwrap().invoice_row;
+        self.inbound_return = Some(inbound_return.clone());
+
+        assert_eq!(inbound_return.r#type, InvoiceRowType::InboundReturn);
```

Comment:
btw, just an option for future, i've been using inline_edit for some test like this:

```
assert_eq!(inbound_return, InvoiceRow {
  r#type: InvoiceRowType::InboundReturn,
  store_id: self.outbound_store.id,
 ..inbound_return.clone()   
})
```

With this method assert_eq would need to be change to a prettier version if error occurs, but it might be cleaner to see what is being tested, so pros and cons of both